src/neuroglancer/util/geom.ts

```diff
@@ -148,3 +149,35 @@ export function translationRotationScaleZReflectionToMat4(
   out[2] = scale[2] * zReflection;
   return mat4.fromRotationTranslationScale(out, rotation, translation, <vec3>temp);
 }
+
+
+/**
+ * Transforms a z-index to/from Morton Codes (i.e. z-indices).
+ */
+const magicBits = [
+  new Uint64(0x49249249, 0x12492492), // AND step
+  new Uint64(0xc30c30c3, 0x30c30c30), // XOR steps
+  new Uint64(0x0f00f00f, 0xf00f00f0),
+  new Uint64(0xff0000ff, 0x00ff0000),
+  new Uint64(0x0000ffff, 0x00ff0000),
+  new Uint64(0x001fffff, 0x00000000),
+];
+
+export function compactMorton(input: Uint64) {
+  let x = input;
+  x = x.and(magicBits[0]);
+  for (let i = 1; i < magicBits.length; ++i) {
+    x = x.xor(x.rshift(Math.pow(2,i))).and(magicBits[i]);
+  }
+  return x;
+}
+
+export function decodeMorton(input: Uint64) {
+  if (input.high) {
```

Comment:
What is the reason for this restriction?  Seems like the morton decoding is designed to handle 64-bit input.